### PR TITLE
Feature/styling

### PR DESCRIPTION
**Pull request by ibugithub**
## Changes Made

- Added Redux Toolkit and Axios dependencies.
- Created async thunks to fetch books, add books, and remove books from the Bookstore API.
- Implemented loading and error states in the Redux store.
- Handled the API responses and updated the state accordingly.
- Updated the UI to display loading indicators and error messages.

## Steps to Test

1. Clone the repository and navigate to the project directory.
2. Install the project dependencies using `npm install`.
3. Start the development server using `npm start`.
4. Open the application in your web browser.
5. Interact with the app by clicking on the fetch books, add book, and remove book buttons.
6. Verify that the loading indicators are displayed appropriately during API requests.
7. Check the console for any error messages.
8. Ensure that the books are fetched, added, and removed correctly from the Bookstore API.

## Checklist

- [ ] I have tested the changes and verified that they work as expected.
- [ ] I have updated the documentation, if necessary.
- [ ] I have added appropriate comments to the code, particularly in complex areas.
- [ ] I have reviewed the code for any potential issues or bugs.
- [ ] I have assigned the PR to the appropriate reviewer.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -99,7 +103,7 @@ You can deploy this project using:
```
Comment (1GORDON) [inline readme.md:107]: Great job @ibugithub 🤝

- [ ] Kindly fix the broken issues link to make your readme file more professional.

File: src/css/Navbar.module.css [hunk 2]
```diff
@@ -4,20 +4,42 @@
...
 
 .nav a:last-child {
   border: unset;
 }
+
+#title {
+  width: 15rem;
+  height: 2.313rem;
+  margin: 0.375rem 2.938rem 0.125rem 0;
+  font-family: Montserrat, sans-serif;
```
Comment (1GORDON) [inline src/css/Navbar.module.css]: Great job @ibugithub 🤝

- [ ] As stated in the project requirements, kindly ensure there is no repeating of the same rule. Instead, create a single class to represent this rule and reuse this class where necessary. For example make sure the following classes are not repeated. `font-family: Montserrat, sans-serif;`, `display: flex`, etc. You may implement this across all your stylesheets. You may use this [article](https://speckyboy.com/good-bad-css-practices/) for more clarity.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @ibugithub ✋

Good job so far!

## Highlights!
- Great use of module CSS✔️ 
- Beautiful UI ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @ibugithub      👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️  
- Nice work with the setup✔️  
  
# Required Changes  🌵 

- [ ] You made some impressive implementations which I must say is a job well done. However, there are a few  I suggest   you make In line with the requirement

- [ ] Kindly resolve the conflict in this branch and make sure your linter files are showing. Its not in the pull request presently


_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @ibugithub    👋

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good job with the use of GitHub features✔️ 
- Kudos on implementing the changes as requested ✅
 
## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**